raft-tech/TANF-app
# Elastic Bulk Document Creation

## Summary of Changes
- Fixed issue where Elastic documents were not being created due to Django bulk model creation not sending appropriate signals
- Bypassed Django signals and manually bulk create Elastic documents for minimized I/O
- This also includes the addition of the Kibana container

## How to Test

```
cd tdrs-frontend && docker-compose up
cd tdrs-backend && docker-compose up --build
```

- Open http://localhost:3000/ and sign in.
- Submit any number of data files
- Verify documents exist in Elastic by running the following command in your terminal: `curl -H "Content-Type: application/json" -X GET "localhost:9200/<INDEX_NAME>/_count"` where <INDEX_NAME> can be `tanf_t1_submissions`.
- Another good verification would be to submit `ADS.E2J.NDM1.TS53_fake.txt` and verify that it takes minutes and not hours to parse/validate. Should be around 3 - 5 minutes.


## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Parsing is still as fast as possible
+ [x] Document creation actually happens now
+ [ ] `adpennington` approved

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
